packages/alfa-rules/src/sia-r15/rule.ts

```diff
@@ -107,3 +106,71 @@ export namespace Outcomes {
     )
   );
 }
+
+type Source = "srcdoc" | "src" | "invalid" | "nothing";
+
+class EmbeddedResource implements Equatable, Hashable {
+  public static of(source: Source, value?: string): EmbeddedResource {
+    return new EmbeddedResource(source, Option.from(value));
+  }
+
+  private readonly _source: Source;
+  private readonly _value: Option<string>;
+
+  private constructor(source: Source, value: Option<string>) {
+    this._source = source;
+    this._value = value;
+  }
+
+  public get value(): Option<string> {
+    return this._value;
+  }
+
+  public equals(value: unknown): value is this {
+    return (
+      value instanceof EmbeddedResource &&
+      this._source === value._source &&
+      this._value.equals(value._value)
```

Comment:
```suggestion
      value._source === this._source &&
      value._value.equals(this._value)
```

Reply:
Side question: is there a reason why we want to have `value` on the left side of comparison (other than unifying with other `equals` in the codebase)?

Reply:
The very first `equals()` method introduced put `value` on the left and `this` on the right as that aligns with the return type: `value is this`.

Reply:
👍 Makes sense.
I was mimicking the order of `this.equals(value)` => `this === value` 😀